features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java

```diff
@@ -104,8 +114,10 @@
     @Autowired
     private SessionUtils sessionUtils;
 
+    private Map<String, Set<BmpPeer>> peersByHashId = new ConcurrentHashMap<>();
```

Comment:
Maybe a `SetMultimap` from guava could help here...

Reply:
Does this needs to be a concurrent map if the method is synchronized?

Reply:
yeah, not needed.  Updated!